cloud/db/services/base.ts

```diff
@@ -327,52 +557,92 @@ export abstract class BaseService<
-      try: async () => {
-        const table = this.getTable();
-        const publicFields = this.getPublicFields();
-        // Type assertion needed: Drizzle's .from() has complex conditional types
-        // (TableLikeHasEmptySelection) that don't work well with generic table types.
-        // Runtime behavior is correct - this is purely a TypeScript limitation.
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const results = await this.db.select(publicFields).from(table as any);
-
-        return results as TPublic[];
-      },
-      catch: (error) =>
-        new DatabaseError({
-          message: `Failed to find all ${this.getResourceName()}s`,
-          cause: error,
-        }),
-    });
-  }
 }
 
-/**
- * Action types for permission checking
- */
+// =============================================================================
+// Authenticated Service
+// =============================================================================
+
+/** CRUD action types used for permission checking. */
 export type PermissionAction = "create" | "read" | "update" | "delete";
 
 /**
- * Permission table mapping actions to allowed roles.
- * Each action maps to an array of roles that can perform that action.
+ * Maps each CRUD action to the roles that can perform it.
+ *
+ * @example
+ * ```ts
+ * const permissions: PermissionTable<"OWNER" | "ADMIN" | "VIEWER"> = {
+ *   create: ["OWNER", "ADMIN"],
+ *   read: ["OWNER", "ADMIN", "VIEWER"],
+ *   update: ["OWNER", "ADMIN"],
+ *   delete: ["OWNER"],
+ * };
+ * ```
  */
 export type PermissionTable<TRole extends string> = Record<
   PermissionAction,
   readonly TRole[]
 >;
 
 /**
- * Abstract base class for services that require authentication.
- * Uses composition with an internal base service for raw CRUD operations.
+ * Abstract base class for services requiring authentication and authorization.
+ *
+ * This class wraps a single `BaseService` instance and adds:
+ * - Role-based permission checking before operations
+ * - Automatic authorization via the `authorize()` helper
+ *
+ * ## Architecture
+ *
+ * ```
+ * ┌───────────────────────────────────────────────┐
+ * │         BaseAuthenticatedService              │
+ * │  ┌───────────────────────────────────────┐    │
+ * │  │   baseService: BaseService<...>       │    │
+ * │  └───────────────────────────────────────┘    │
+ * │                                               │
+ * │  + getRole()             // Determine role    │
+ * │  + getPermissionTable()  // Define roles      │
+ * │  + authorize()           // Check permissions │
+ * └───────────────────────────────────────────────┘
+ * ```
+ *
+ * ## Implementing an Authenticated Service
+ *
+ * Subclasses must implement:
+ * - `initializeBaseService()` - Create and return the BaseService instance
+ * - `getPermissionTable()` - Define which roles can perform which actions
+ * - `getRole()` - Determine the authenticated user's role for a resource
```

Comment:
nit: comments like `getRole()` imply that getRole takes no arguments, which is misleading. I suggest either something like `getRole(args)` or perhaps more detailed arg structure for further clarity.